web-ui/components/Staking/modals/stakingProcessModal.tsx

```diff
@@ -209,32 +203,54 @@ export const StakingProcessModal: React.FC<StakingModalProps> = ({ isOpen, onClo
   const { data: zone, isLoading: isZoneLoading } = useZoneQuery(selectedOption?.chainId ?? '');
 
   const valToByte = (val: number) => {
-    if (val > 1) {
-      val = 1;
-    }
-    if (val < 0) {
-      val = 0;
-    }
+    if (val > 1) val = 1;
+    if (val < 0) val = 0;
     return Math.abs(val * 200);
   };
 
+  // Function to add validator to memo
   const addValidator = (valAddr: string, weight: number) => {
-    let { words } = bech32.decode(valAddr);
-    let wordsUint8Array = new Uint8Array(bech32.fromWords(words));
-    let weightByte = valToByte(weight);
+    const { words } = bech32.decode(valAddr);
+    const wordsUint8Array = new Uint8Array(bech32.fromWords(words));
+    const weightByte = valToByte(weight);
+
     return Buffer.concat([Buffer.from([weightByte]), wordsUint8Array]);
   };
 
+  // Function to add mapped address to memo
+  const addMappedAccount = (mappedAddr: string) => {
+    if (!mappedAddr) {
+      return Buffer.alloc(0);
+    }
+    const { words } = bech32.decode(mappedAddr);
+    const wordsUint8Array = new Uint8Array(bech32.fromWords(words));
+
+    if (wordsUint8Array.length !== 20) {
+      throw new Error('Mapped address must be 20 bytes long');
+    }
+
+    // 0x00 for field type, 0x14 for length, then the address bytes
+    return Buffer.concat([Buffer.from([0x00, 0x14]), wordsUint8Array]);
+  };
+
+  // Create the memo
   let memoBuffer = Buffer.alloc(0);
 
+  // Add validator details to memo
   if (intents.length > 0) {
     intents.forEach((val) => {
       memoBuffer = Buffer.concat([memoBuffer, addValidator(val.address, val.intent)]);
     });
+    // Add mapped account to memo
+    if (zone?.chainId === 'agoric-3' && address) {
+      memoBuffer = Buffer.concat([memoBuffer, addMappedAccount(address)]);
+    }
+    // Add header with the memo buffer length
```

Comment:
This needs to be after line 249. The 0x02 <length> prefix pertains to the intents field.

Lines 244-248 should not be inside the `if (intents.length > 0) {` condition.

Reply:
apologies for the oversight!